buildengine/engine.go

```diff
@@ -350,48 +352,36 @@ func (e *Engine) buildAndDeploy(ctx context.Context, replicas int32, waitForDepl
 		projects = maps.Keys(e.projects)
 	}
 
-	deployQueue := make(chan Project, len(projects))
-	wg, ctx := errgroup.WithContext(ctx)
+	buildGroup, buildCtx := errgroup.WithContext(ctx)
+	deployGroup, deployCtx := errgroup.WithContext(ctx)
```

Comment:
If your goal is to avoid cancelling the context on failure, don't use `errgroup.WithContext()` - instead create an errgroup manually like so `wg := errgroup.Group{}`